### PR TITLE
optimize: reduce the number of lock conflict exception

### PR DESCRIPTION
**Pull request by ggndnn**
### Ⅰ. Describe what this PR did
Move lock retry loop into ConnectionProxy.commit, that the number of LockKeyConflict can be significantly reduced whether auto-commit is true or false. Related issue #1438.

### CODE REVIEW

**Comment by slievrly** [comment]
@ggndnn I am very interested because we always have different ideas. When a LockConflictException occurs, it indicates that there are other distributed transactions that are executing that hold the same data primary key. We define the current distributed transaction as A and another distributed transaction as B. A holds the database lock and wants to get the global lock, and B holds the global lock. If at this point B wants to roll back this data in the second phase of the distributed transaction, it will try to acquire the database lock. According to your code, A will hold the database lock for a longer time. At this time, B may trigger the lock wait timeout exception and perform a rollback retry. We need to evaluate this.

**Comment by ggndnn** [comment]
@slievrly It's indeed interesting to have discussions with you, the atmosphere here is good， different ideas always make things more clear. Thank you. 

Before I make this PR, I referred to seata document in wiki, e.g.  

I think tx2 is A which you just mentioned and tx1 is B. I understand your worries. I also agree that we should make `quick fail` to avoid waiting too much. But `rollback` is not what we preferred, we prefer `commit`, according to issue 1438, we got exception immediately, may be just a moment we can get a success transaction. I think this PR is more in line with the figure in wiki. Is that a formal design?

I agree that we should evaluate this.  What can we do with this PR?

**Comment by slievrly** [comment]
@ggndnn I am very happy to discuss this with you. I understand what you mean above. By the way, is it appropriate for us to communicate through DingTalk?

**Comment by ggndnn** [comment]
@slievrly I joined DingTalk group.

**Comment by slievrly** [comment]
@ggndnn Do you want to add a switch here as we communicate?

**Comment by ggndnn** [comment]
@slievrly Ok, I'll add a switch.

**Comment by ggndnn** [comment]
@slievrly I added a switch. Please have a look whether it's appropriate.

File: rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java [hunk 1]
```diff
@@ -83,34 +84,23 @@ protected T executeAutoCommitFalse(Object[] args) throws Exception {
...
-
+            return new LockRetryPolicy(connectionProxy.getTargetConnection()).execute(() -> {
+                T result = executeAutoCommitFalse(args);
+                connectionProxy.commit();
+                return result;
+            });
         } catch (Exception e) {
             // when exception occur in finally,this exception will lost, so just print it here
             LOGGER.error("execute executeAutoCommitTrue error:{}", e.getMessage(), e);
+            connectionProxy.rollback();
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java]: just used for LOCK_RETRY_POLICY_BRANCH_ROLLBACK_ON_CONFLICT=false

Comment (ggndnn) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java]: Should it be okay to do `ConnectionProxy#rollback` in both cases?

Comment (slievrly) [reply rm-datasource/src/main/java/io/seata/rm/datasource/exec/AbstractDMLBaseExecutor.java]: > Should it be okay to do `ConnectionProxy#rollback` in both cases?

there is no problem. I just mark ,when   LOCK_RETRY_POLICY_BRANCH_ROLLBACK_ON_CONFLICT=true ,it  already rollback , never run here.

**Pull request merged**